### DISCUSSION TITLE
Can we have `i_Ctrl-k` for digraphs & `i_Ctrl-h` for `signature_help()`?

### DISCUSSION THREAD

**Discussion by txtyash** [LazyVim/LazyVim#361] [General]
`i_Ctrl-k` is the default binding for digraphs which is being used for signature help. Can we revert? And use `i_Ctrl-h` for it instead. This would only break the Backspace behavior that's possible with `i_Ctrl-h` for which we have the `<Backspace>` key so it wouldn't be a problem... :confused:

**Comment by txtyash** [comment]
Or at least guide me remap `i_ctrl-h` for signature help & unmap `i_ctrl-k` :sweat_smile: 
```lua
map("i", "<C-h>", vim.lsp.buf.signature_help(), { desc = "Copy entire file" })
```

**Reply by dpetka2001** [reply]
Please see [this](https://www.lazyvim.org/plugins/lsp#%EF%B8%8F-customizing-lsp-keymaps) for customizing LSP mappings.

**Reply by txtyash** [reply]
There's no way to specify an `insert` mode binding here?
```lua
    local keys = require("lazyvim.plugins.lsp.keymaps").get()
    -- change a keymap
    keys[#keys + 1] = { "K", "<cmd>echo 'hello'<cr>" }
    -- disable a keymap
    keys[#keys + 1] = { "K", false }
   ```

**Reply by txtyash** [reply]
And trying to disable `i_Ctrl-k` didn't work for me:
```lua
return {
  {
    "neovim/nvim-lspconfig",
    init = function()
      local keys = require("lazyvim.plugins.lsp.keymaps").get()
      -- Disable i_Ctrl-k as it is used for digraphs
      keys[#keys + 1] = { "<C-k>", false }
    end,
  },
}
```

**Reply by txtyash** [reply]
Neither did this:
```lua
return {
  {
    "neovim/nvim-lspconfig",
    keys = {
      -- Disable i_Ctrl-k as it is used for digraphs
      { "<C-k>", false },
    },
  },
}
```

**Reply by folke** [reply]
> And trying to disable `i_Ctrl-k` didn't work for me:
> ```lua
> return {
>   {
>     "neovim/nvim-lspconfig",
>     init = function()
>       local keys = require("lazyvim.plugins.lsp.keymaps").get()
>       -- Disable i_Ctrl-k as it is used for digraphs
>       keys[#keys + 1] = { "<C-k>", false }
>     end,
>   },
> }
> ```

Add mode="i"

**Reply by txtyash** [reply]
nope, didn't work.
```lua
keys[#keys + 1] = { "<C-k>", false, mode = "i" }
```
but this worked:
```lua
keys[#keys + 1] = { "<C-h>", vim.lsp.buf.signature_help, mode='i' }
```

**Reply by folke** [reply]
lower-case `<c-k>` and with `mode = "i"`

**Reply by txtyash** [reply]
That worked!
